roles/run_hook/tasks/main.yml

```diff
@@ -14,8 +14,41 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-- name: Loop on the passed hooks and call correct action
-  ansible.builtin.include_tasks: "{{ hook.type }}.yml"
-  loop: "{{ hooks }}"
-  loop_control:
-    loop_var: 'hook'
+- name: "Combine hooks and run them for {{ step }}"
+  vars:
+    _list_hooks: >-
+      {{
+        hostvars[inventory_hostname][step] |
+        default(vars[step] | default([]))
+      }}
+    _matcher: "^{{ step }}_(.*)$"
+    _filtered_hooks: >-
+      {{
+        hostvars[inventory_hostname] |
+        dict2items |
+        selectattr('key', 'match', _matcher)
+      }}
+    _single_hooks: >-
+      {% for hook in _filtered_hooks -%}
+      - name: {{ hook.key | regex_replace(_matcher, '\1') |
+                 replace('_', '') | capitalize }}
+        {{ hook.value | to_nice_yaml | indent(2) }}
+      {% endfor -%}
+    _hooks: >-
```

Comment:
it seems ansible is still passing some string in here instead of actual list - need to find out what, and why.